R/pkg/inst/worker/worker.R

```diff
@@ -84,6 +84,10 @@ broadcastElap <- elapsedSecs()
 # as number of partitions to create.
 numPartitions <- SparkR:::readInt(inputCon)
 
+# If true, working for RDD
+# If false, working for DataFrame
+isDataFrame <- as.logical(SparkR:::readInt(inputCon))
```

Comment:
This is a good point. I think the assumption is that R worker processes are started using the same binary release as the JVM processes. But yeah having a protocol version number or something like that might be interesting to explore.